libp2p/protocols/protocol.nim

```diff
@@ -26,11 +30,16 @@ type
     codecs*: seq[string]
     handler*: LPProtoHandler ## this handler gets invoked by the protocol negotiator
     started*: bool
+    maxStreams*: Opt[int]
```

Comment:
Can't this be an `int` with a default value of `DefaultMaxStreams`?

Reply:
Default values for object just landed for nim 2.0: https://github.com/nim-lang/Nim/pull/20480
So we will probably be able to use them in 2 years, once we drop support for 1.6

In the meantime, that's the best way to do this, imo (other solution would be to use `0` as a default = `10`, but I find this not great)

Reply:
can't we use a new proc?

Reply:
We would have to update every protocol to call this `new`, but some protocols are in the users codebase, so we can't :/